### ISSUE TITLE
patch-package failed but git apply succeed

### ISSUE THREAD

**Issue by leethree** [ds300/patch-package#22]
We are using `patch-package` for developing our app and it works great in development environments. But it shows a strange error when we deployed it to CircleCI. 

```
$ "/home/circleci/project/node_modules/.bin/patch-package"
patch-package: Applying patches...

**ERROR** Failed to apply patch for package react-native-device-info

  This error was caused because Git cannot apply the following patch file:

    patches/react-native-device-info+0.11.0.patch

  This is usually caused by inconsistent whitespace in the patch file.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Exited with code 1
```

But the patches seem to be working fine when I use `git apply`
```
$ git apply patches/*
```
(Succeeded, no output message)

The environment for CircleCI is:
- Docker container: [circleci/node:8](https://hub.docker.com/r/circleci/node/)
- Node v8.6.0
- Yarn v1.1.0
- patch-package v3.4.2

**Comment by ds300** [comment]
Thanks for the report, I'm looking into it!

**Comment by ds300** [comment]
I managed to reproduce the issue. It's caused because the `--unsafe-paths` option we use for `git apply` is not supported with the version of git provided on that docker image. It has 2.1.4, but `--unsafe-paths` first appeared around 2.3 (from January 2015). Is it possible for you to use a docker image with a newer version of Git?

**Comment by ds300** [comment]
I'm going to fix this anyway. IIRC `--unsafe-paths` is only useful for situations where the project is not under source control with `git`, so I'll just avoid using it unless that's the case.

**Comment by ds300** [comment]
Hmm, so `--unsafe-paths` is actually only useful for situations where the files to be patched are outside of the current working directory, which I think should never happen with patch-package. I'll try just removing it altogether.

**Comment by ds300** [comment]
This should be fixed now in patch-package@3.4.4 - feel free to reopen the issue if you find otherwise.

**Comment by leethree** [comment]
I confirm the issue is fixed with patch-package@3.4.4. It now works great on the CircleCI image. Thank you a lot!
